Parametric/Denotation/MEvaluation.agda

```diff
@@ -0,0 +1,70 @@
+------------------------------------------------------------------------
+-- INCREMENTAL λ-CALCULUS
+--
+-- Standard evaluation for MTerm
+------------------------------------------------------------------------
+
+import Parametric.Syntax.Type as Type
+import Parametric.Syntax.Term as Term
+import Parametric.Denotation.Value as Value
+
+import Parametric.Syntax.MType as MType
+import Parametric.Syntax.MTerm as MTerm
+import Parametric.Denotation.MValue as MValue
+
+
+module Parametric.Denotation.MEvaluation
+    {Base : Type.Structure}
+    (Const : Term.Structure Base)
+    (⟦_⟧Base : Value.Structure Base)
+    (ValConst : MTerm.ValConstStructure Const)
+    (CompConst : MTerm.CompConstStructure Const)
+  where
+
+open Type.Structure Base
+
+open MType.Structure Base
+open MTerm.Structure Const ValConst CompConst
+open MValue.Structure Base ⟦_⟧Base
+
+open import Base.Denotation.Notation
+
+-- Extension Point: Evaluation of fully applied constants.
+ValStructure : Set
+ValStructure = ∀ {Σ τ} → ValConst Σ τ → ⟦ Σ ⟧ → ⟦ τ ⟧
+
+CompStructure : Set
+CompStructure = ∀ {Σ τ} → CompConst Σ τ → ⟦ Σ ⟧ → ⟦ τ ⟧
+
+module Structure
+       (⟦_⟧ValBase  : ValStructure)
+       (⟦_⟧CompBase : CompStructure)
+    where
+
+  -- We provide: Evaluation of arbitrary value/computation terms.
+  ⟦_⟧Comp : ∀ {τ Γ} → Comp Γ τ → ⟦ Γ ⟧ValContext → ⟦ τ ⟧CompType
+  ⟦_⟧Val : ∀ {τ Γ} → Val Γ τ → ⟦ Γ ⟧ValContext → ⟦ τ ⟧ValType
+  ⟦_⟧Vals : ∀ {Γ Σ} → Vals Γ Σ → ⟦ Γ ⟧ → ⟦ Σ ⟧
+
+  ⟦ ∅ ⟧Vals ρ = ∅
+  -- XXX I don't use this enough, but Val should be ValTerms or ValTms (since
```

Comment:
This is a naming question. @Toxaris, do you have opinions? If you're confused, I guess I can most easily explain in person.